lib/Sidebar/Sidebar.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { node, bool, oneOf, string } from 'prop-types';
+import cx from 'classnames';
+import { useTransition, animated } from 'react-spring';
+
+export function Sidebar({ children, isOpen, direction, width, className }) {
+  const classNames = cx(
+    {
+      'border-l border-r-0': direction === 'right',
+      'border-r border-l-0': direction === 'left'
+    },
+    'border-solid border-neutral-90 border-b-0 border-t-0 h-full bg-white shadow',
+    className
+  );
+
+  const transitions = useTransition(isOpen, null, {
+    from: {
+      position: 'absolute',
+      width,
+      [direction]: `-${width}`
+    },
+    enter: {
+      [direction]: '0'
+    },
+    leave: {
+      [direction]: `-${width}`
+    }
+  });
```

Comment:
I had the biggest issue with `width`. The width of the sidebar is required because it then knows how far to animate to get the whole sidebar showing. I had thought about not passing in the width, and instead using ref and getting it's width. But that felt like a lot effort.

Instead we can just pass in a width, which can be `px` or `%`. It can't be `rem` or `em` because the `enter` value needs to be the same type! I think this approach is okay because currently we only intend to pass in a `%`, just enough `%` to cover the existing inline comments.